src/format/opamTypes.mli

```diff
@@ -273,21 +284,14 @@ type user_action =
 
 (** Solver universe *)
 type universe = {
+  u_pefuniv  : ((string * string), Opam.Packages.package) Hashtbl.t;
+  u_options  : (string * string list * string list);
+  u_tables   : Pef.Pefcudf.tables;
```

Comment:
This is great, as it means we won't have to reload different kinds of universes, doing lots of duplicate conversions in the way, just to compute the different dependency graphs. Also, even at the upper level, we could load the universe just once and pass it along, specifying the action, doc, build and test flags just when we need a specific dependency graph (in the current setting, it could be part of the switch state, but for multi-switch, we could have it along the switch-name=>switch-state map in a new `multiswitch-state` structure).

I am not fond of using strings for flags, but not sure what the best interface would be. Maybe keep the higher-level types at this point, and do the conversion in the interface with Dose ? Or just make this more opaque to the higher-level, since we should only need to load once ? Well, that's mostly aesthetic concerns anyway.

Something nice we could do is actually get rid of the `doc` and `tests` flags at the lower level (they'll never change by then, they're defined globally, and even if we wanted to define them per-package we would do this at the upper level). The two "dependency toggles" that we really need are actually `build` and `depopt`, which play a similar role despite being defined completely differently. So if we could have a single universe in Dose, with these two toggles on dependencies, that would be awesome and result in big clarity and performance gains, by making the factorisation of most of the translations possible. What do you think ?